src/librustc_typeck/check/intrinsic.rs

```diff
@@ -78,6 +78,9 @@ pub fn check_intrinsic_type<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>,
                                       it: &hir::ForeignItem) {
     let param = |n| tcx.mk_param(n, Symbol::intern(&format!("P{}", n)).as_interned_str());
     let name = it.name.as_str();
+    let va_list_ty = tcx.lang_items().va_list().map(|did| {
+        tcx.type_of(did)
+    });
```

Comment:
Okay, I took a look at the rest of this file, and I think the fix for the lifetime problem is:
https://github.com/rust-lang/rust/blob/3c31e17ddc8914365145683e6d56fa0229a01a8e/src/librustc_typeck/check/intrinsic.rs#L122-L123
Specifically, the `mk_region` call, which could be used like so:
```rust
        tcx.type_of(did).subst(tcx, &[
            tcx.mk_region(ty::ReLateBound(ty::DebruijnIndex::new(1), ty::BrAnon(0))).into()
        ])
```

Reply:
Ah! That is awesome! I don't know why I didn't think to use the `subst`. Great catch.